### PR TITLE
4 create post comment backend

### PR DESCRIPTION
**Pull request by islamsaied**
closes #5

### CODE REVIEW

**Comment by aymanmoukhtar** [comment]
Please create an issue, and put its number as a reference in the PR comment
If you write `closes #issueNumber`, the issue will be closed automatically when this PR is merged

File: Backend/Linkedout.Domain/Enums/ReactOnTypeEnum.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Linkedout.Domain.Enums
+{
+    public enum ReactOnTypeEnum
```
Comment (aymanmoukhtar) [inline Backend/Linkedout.Domain/Enums/ReactOnTypeEnum.cs:7]: Enums, shouldn't have `Enum` in their name. It should be `ReactOnType`

File: Backend/Linkedout.Domain/Enums/ReactionTypeEnum.cs [hunk 2]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Linkedout.Domain.Enums
+{
+    public enum ReactionTypeEnum
```
Comment (aymanmoukhtar) [inline Backend/Linkedout.Domain/Enums/ReactionTypeEnum.cs:7]: Should be `ReactionType`

File: Backend/Linkedout.Domain/Interfaces/Repository/ICommentRepository.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿using Linkedout.Domain.Entities;
+using Linkedout.Domain.Entities.Posts;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Linkedout.Domain.Interfaces.Repository
+{
+    public interface ICommentRepository : IRepository<Comment>
```
Comment (aymanmoukhtar) [inline Backend/Linkedout.Domain/Interfaces/Repository/ICommentRepository.cs:9]: I can't see it used anywhere, same idea as the `IPostRepository`

File: Backend/Linkedout.Domain/Interfaces/Repository/IPostRepository.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using Linkedout.Domain.Entities;
+using Linkedout.Domain.Entities.Posts;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Linkedout.Domain.Interfaces.Repository
+{
+    public interface IPostRepository : IRepository<Post>
```
Comment (aymanmoukhtar) [inline Backend/Linkedout.Domain/Interfaces/Repository/IPostRepository.cs:9]: Not used.

File: Backend/Linkedout.Infrastructure/Repository/PostRepository.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+using Linkedout.Domain.Interfaces.Repository;
+using Linkedout.Infrastructure.Repository.SqlServerContext;
+using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Text;
+
+namespace Linkedout.Infrastructure.Repository
+{
+    public class PostRepository : SqlServerRepository<Post>, IPostRepository
```
Comment (aymanmoukhtar) [inline Backend/Linkedout.Infrastructure/Repository/PostRepository.cs:12]: Not Used

File: Backend/Linkedout/GraphQL/Queries/UserQueries.cs [hunk 6]
```diff
@@ -19,7 +19,7 @@ Lazy<IMediator> mediator
             _mediator = mediator;
         }
 
-        [Authorize]
```
Comment (aymanmoukhtar) [inline Backend/Linkedout/GraphQL/Queries/UserQueries.cs:22]: Is there a reason for removing it?

File: Backend/Linkedout/GraphQL/Queries/PostQueries.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        private readonly Lazy<IMediator> _mediator;
+
+        public PostQueries(
+            Lazy<IMediator> mediator
+            )
+        {
+            _mediator = mediator;
+        }
+
+        public async Task<List<PostViewModel>> GetPostsByUserId(string userId) => await _mediator.Value.Send(new GetPostsByUserIdQuery { UserId = userId });
```
Comment (aymanmoukhtar) [inline Backend/Linkedout/GraphQL/Queries/PostQueries.cs:22]: It should have the [Authorized] attribute

File: Backend/Linkedout/GraphQL/Mutations/PostMutations.cs [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+        private readonly Lazy<IMediator> _mediator;
+
+        public PostMutations(
+            Lazy<IMediator> mediator
+            )
+        {
+            _mediator = mediator;
+        }
+
+        public async Task<Post> CreatePost(CreatePostViewModel post) => await _mediator.Value.Send(new CreatePostCommand { Post = post });
```
Comment (aymanmoukhtar) [inline Backend/Linkedout/GraphQL/Mutations/PostMutations.cs:24]: Should have the [Authorized] attribute

File: Backend/Linkedout.Application/Post/Queries/GetPostsByUserId/GetPostsByUserIdQuery.cs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+                Content = _.Content,
+                ReactionsCount = _.ReactionsCount,
+                Creator = new UserViewModel
+                {
+                    Id = _.Creator.Id,
+                    FirstName = _.Creator.FirstName,
+                },
+            }).ToListAsync();
+
+            return result;
```
[the remark this answers is not in the document]
Comment (islamsaied) [reply Backend/Linkedout.Application/Post/Queries/GetPostsByUserId/GetPostsByUserIdQuery.cs:44]: it's more usable in debugging with multiple lines shape

[the remark this answers is not in the document]
Comment (aymanmoukhtar) [reply Backend/Linkedout.Application/Post/Queries/GetPostsByUserId/GetPostsByUserIdQuery.cs:44]: Agreed, but the final version of production code should be as concise and memory efficient as possible.
This one is a very minor one, but embracing that style is ultimately useful

File: Backend/Linkedout/GraphQL/Queries/UserQueries.cs (hunk 6 above)
Comment (islamsaied) [reply Backend/Linkedout/GraphQL/Queries/UserQueries.cs:22]: Just for testing and forget to put it again

File: Backend/Linkedout.Application/Post/Queries/GetPostsByUserId/GetPostsByUserIdQuery.cs [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    using Linkedout.Domain.ViewModels.User;
+
+    public class GetPostsByUserIdQuery : IRequest<List<PostViewModel>>
+    {
+        public string UserId { get; set; }
+    }
+
+    public class GetPostsByUserIdQueryHandler : IRequestHandler<GetPostsByUserIdQuery, List<PostViewModel>>
+    {
+        private readonly Lazy<IReadonlyRepository<Post>> _postReadonlyRepository;
```
[the remark this answers is not in the document]
Comment (islamsaied) [reply Backend/Linkedout.Application/Post/Queries/GetPostsByUserId/GetPostsByUserIdQuery.cs]: As per our conversation we confirmed using of Generic Repo unless any business logic forced us to build customize Repo for an specific entity

**Pull request merged**